index.html

```diff
@@ -871,54 +868,132 @@ <h4>
+          for (let record of socialPost.toRecords()) {
+            switch (record.recordType) {
+              case "text":
+                text = record.toText();
+                break;
+              case "-act":
+                const buffer = record.toArrayBuffer();
+                const view = new DataView(buffer);
+                action = view.getUint8(0);
+                break;
+            }
+          }
+
+          switch (action) {
+            case 0: // do the action
+              console.log(`Post "${text}" to timeline`);
+              break;
+            case 1: // save for later
+              console.log(`Save "${text}" as a draft`);
+              break;
+            case 2: // open for editing
+              console.log(`Show editable post with "${text}"`);
+              break;
+          }
+        };
 
-    <pre class="highlight">
-  const reader = new NFCReader();
-  reader.onreading = event => {
-    const socialPost = event.message.records[0];
-    if (!socialPost) {
-      return;
-    }
-
-    let action;
-    let text = "";
-
-    for (let record of socialPost.toRecords()) {
-      switch (record.recordType) {
-        case "text":
-          text = record.toText();
-          break;
-        case "-act":
-          const buffer = record.toArrayBuffer();
-          const view = new DataView(buffer);
-          action = view.getUint8(0);
-          break;
-      }
-    }
-
-    switch (action) {
-      case 0: // do the action
-        console.log(`Post "${text}" to timeline`);
-        break;
-      case 1: // save for later
-        console.log(`Save "${text}" as a draft`);
-        break;
-      case 2: // open for editing
-        console.log(`Show editable post with "${text}"`);
-        break;
-    }
-  };
-
-  reader.scan({ recordType: "example.com:sp"});
+        reader.scan({ recordType: "example.com:sp"});
+      </pre>
+    </section>
+
+    <section> <h3>Push a smart poster message</h3>
+      <p>
+        Smart poster is defined in the NFC Forum Smart Poster Record Type
+        Definition specification as an NDEF record that contains an
+        <a>NDEF message</a> as payload, which may contain several records: a
+        mandatory URI record that refers to a content, and additional optional
+        records for that content's title, icon, size, type and possible action
+        (e.g. the value `0` for doing an action associated to the content, like
+        send SMS, make call, launch browser, or the value `1` for saving the
+        content, or the value `2` for opening for editing).
+      </p>
+      <pre class="example">
+        const writer = new NFCWriter();
+        writer.push({ records: [
+          {
+            recordType: "smart-poster",
+            data: { records: [
+              {
+                recordType: "url",
+                data: "https://my.org/content/19911"
+              },
+              {
+                recordType: "t", // smart poster type, a local type to Sp
```

Comment:
How are these local types exposed when reading the records? `smart-poster:t` ?

Anyway we can look at that in follow ups

Reply:
According to the Sp RTD it's just the "t" for title and "act" for action, then the regular tnf=1 text, tnf=1 uri, and tnf=2 (mime type).

Reply:
yes but these are local, so that t is restricted to Sp records, but I guess that is fine exposing it like that